docs/rfc/0004_vulnerability_report.md

```diff
@@ -0,0 +1,245 @@
+	V3Score  float64 `json:"v3score,omitempty"`
+}
+```
+
+`VexStatus` reports information about VEX only when the 
+`Vulnerability.Suppressed` is `true`, otherwise this is empty.
+
+* `Repository` is the VEX repo used to suppress it.
+* `Status` is the status of the vulnerability such as `not_affected`, 
+`affected`, `fixed`, or `under_investigation`.
+* `Statement` explains more about the status 
+(eg. `vulnerable_code_not_present`).
+
+```golang
+type VexStatus struct {
+	Repostory string `json:"repository,omitempty"`
+	Status    string `json:"status"`
+	Statement string `json:"statement"`
+}
+```
+
+`Vulnerability` holds information about the target exposure.
+
+* `CVE` is the Common Vulnerabilities and Exposures identifier 
+(e.g., CVE-2024-1234). It uniquely identifies the vulnerability in public 
+databases.
+* `PackageName` is the name of the affected software package where the 
+vulnerability was found.
+* `PackagePath` is the path to the package in the file system or within a 
+container layer. Useful for locating the affected component in complex projects.
+* `PURL` (Package URL) is a standardized way to describe and identify the 
+software package, useful for correlation across tools and ecosystems.
+* `InstalledVersion` is the version of the package that is currently installed 
+and affected by the vulnerability.
+* `FixedVersions` is a list of versions where the vulnerability has been 
+addressed. Helps determine what upgrade paths are available.
+* `LayerDiffID` is the unique identifier of the container image layer that 
+introduced the package. Useful for tracing the origin of the vulnerability 
+in layered images.
+* `Description` provides a human-readable explanation of the vulnerability.
+* `Severity` indicates the criticality level of the vulnerability 
+(e.g., LOW, MEDIUM, HIGH, CRITICAL).
+* `Cvss` contains the CVSS (Common Vulnerability Scoring System) score and 
+vector, used to quantify the severity of the vulnerability.
+* `References` is a list of URLs or documentation links for further reading 
+or remediation steps.
+* `Suppressed` indicates whether the vulnerability has been suppressed 
+(by a VEX document).
+* `VexStatus` provides the VEX (Vulnerability Exploitability eXchange) status.
+
+```golang
+type Vulnerability struct {
+	CVE              string   `json:"cve"`
+	PackageName      string   `json:"packageName"`
+	PackagePath      string   `json:"packagePath,omitempty"`
+	PURL             string   `json:"purl"`
+	InstalledVersion string   `json:"installedVersion"`
+	FixedVersions    []string `json:"fixedVersions,omitempty"`
+	LayerDiffID      string   `json:"layerDiffId,omitempty"`
+	Description      string   `json:"description,omitempty"`
+	Severity         string   `json:"severity,omitempty"`
+	Cvss             CVSS     `json:"cvss"`
+	References       []string `json:"references,omitempty"`
+	Suppressed       bool      `json:"suppressed"`
+	VexStatus        VexStatus `json:"vexStatus,omitempty"`
+}
+```
```

Comment:
LGTM!
One small note: Trivy and NeuVector both include `PublishedDate` and `LastModifiedDate`, and Trivy also carries a `DataSource` block such as:

```json
"DataSource": {
  "ID": "debian",
  "Name": "Debian Security Tracker",
  "URL": "https://salsa.debian.org/security-tracker-team/security-tracker"
}
```

Because we already expose `pr_url`, I think that should be sufficient.